### ISSUE TITLE
shutdownAllBrowsers doesn't call back if browser.shutdown errors

### ISSUE THREAD

**Issue by vweevers** [airtap/airtap#93]
Because it only listens for the `done` event. Would be nicer if we can pass a callback to `browser.shutdown`. @ralphtheninja WDYT?

https://github.com/airtap/airtap/blob/54d99c2cc32174f832548ab208586b247d6071b5/bin/airtap.js#L291-L304

Also, move the `require('batch')` out of the function.

**Comment by vweevers** [comment]
I'm not familiar with `batch` though, should check how it handles errors. I.e. if an error aborts the batch (in this case, it should not).

**Comment by ralphtheninja** [comment]
Yes, I think that's a good idea. The thought had struck my mind, and I originally played around with passing a callback to phantom and electron browsers, but realized the signature for sauce browser was different.

**Comment by ralphtheninja** [comment]
We would have to rewrite `SauceBrowser.js` a bit. We could check if `err` is an error or a callback though (in `.shutdown`), but slightly ugly.

**Comment by vweevers** [comment]
The whole thing is awkward but we have to start somewhere :) I think this is acceptable:

```js
FooBrowser.prototype.shutdown = function (err, callback) {
   if (typeof err === 'function') {
      callback = err
      err = null
   }

   // ..
}
```

**Comment by vweevers** [comment]
Related issue: `SauceBrowser` doesn't pass a callback to `controller.shutdown`:

https://github.com/airtap/airtap/blob/3360313d135466b4e3db828d4097e1f578c35930/lib/SauceBrowser.js#L252-L254

Though `shutdown` does accept one:

https://github.com/airtap/airtap/blob/3360313d135466b4e3db828d4097e1f578c35930/lib/setup.js#L96-L102

And we may want to add something like [server-destroy](https://github.com/isaacs/server-destroy) to our server(s) to close active connections.

**Comment by vweevers** [comment]
And the errors of PhantomJS and Electron are swallowed:

https://github.com/airtap/airtap/blob/3360313d135466b4e3db828d4097e1f578c35930/lib/airtap.js#L92-L97

I'm thinking we should remove the `done` and `error` events altogether, and eventually refactor it to:

```js
browser.start(done)
```

**Comment by ralphtheninja** [comment]
> And we may want to add something like server-destroy to our server(s) to close active connections.

Didn't know about this module. Very handy!

**Comment by ralphtheninja** [comment]
> I'm thinking we should remove the done and error events altogether, and eventually refactor it to a pattern like:
> 
> browser.start(done)

+1

**Comment by vweevers** [comment]
Not sure how to attack all this in clean PR(s), so I'm gonna play around a bit.
